logistics_project/apps/malawi/warehouse/report_views/stock_status.py

```diff
@@ -72,70 +73,97 @@ def get_stock_status_by_product_table(self, reporting_supply_point, is_facility)
             "data": status_data,
         }
 
-    def custom_context(self, request):
-        selected_type = self.get_selected_product_type(request)
-        selected_product = self.get_selected_product(request)
-        reporting_supply_point = self.get_reporting_supply_point(request)
-
-        headings = ["% HSA Stocked Out", "% HSA Under", "% HSA Adequate", 
-                    "% HSA Overstocked", "% HSA Not Reported"]
-
-        hsa_table = None
-        location_table = {
-            "id": "location-table",
+    def get_months_of_stock_table(self, reporting_supply_point, is_facility):
+        products = Product.objects.filter(is_active=True, type__is_facility=request.is_facility).order_by('sms_code')
+        table = {
+            "id": "months-of-stock-table",
             "is_datatable": True,
             "is_downloadable": True,
+            "data": [],
+            "location_type": "Facility" if is_facility else "HSA"
         }
-            
-        if is_facility(reporting_supply_point):
-            products = Product.objects.filter(is_active=True, type__is_facility=request.is_facility).order_by('sms_code')
-            hsa_table = {
-                "id": "hsa-table",
-                "is_datatable": True,
-                "is_downloadable": True,
-                "data": [],
-                "location_type": "HSA"
-            }
-            
-            hsa_table["header"] = [hsa_table["location_type"]] + \
-                [p.sms_code for p in products]
-            
-            # this chart takes a long time to load
-            hsas = hsa_supply_points_below(reporting_supply_point.location)
-            for hsa in hsas:
-                temp = [hsa.name]
-                for product in products:
-                    ps = ProductStock.objects.filter(supply_point=hsa, product=product)
-                    if ps.count():
-                        mr = ps[0].months_remaining
-                        if mr:
-                            temp.append('%.1f' % mr)
-                        else:
-                            temp.append('-')
+
+        table["header"] = (
+            [table["location_type"]] +
+            [p.sms_code for p in products]
+        )
+
+        if is_facility:
```

Comment:
:+1: yeah I like the `base_level` idea and that sounds like it will address the bulk of the confusion. sounds good to me.